src/core/Geometry.js

```diff
@@ -988,7 +988,20 @@ Geometry.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 
 			for ( var key in parameters ) {
 
-				if ( parameters[ key ] !== undefined ) data[ key ] = parameters[ key ];
+				var value = parameters[ key ];
```

Comment:
Instead of shading `Geometry`, why not implement `TubeBufferGeometry.toJSON()`. In this way, the code is similar to `ExtrudeBufferGeometry` and `ShapeBufferGeometry`.